app/src/main/java/com/eveningoutpost/dexdrip/Services/MissedReadingService.java

```diff
@@ -42,11 +43,17 @@ protected void onHandleIntent(Intent intent) {
         bg_missed_minutes =  Integer.parseInt(prefs.getString("bg_missed_minutes", "30"));
         otherAlertSnooze =  Integer.parseInt(prefs.getString("other_alerts_snooze", "20"));
 
-        if (bg_missed_alerts && BgReading.getTimeSinceLastReading() > (bg_missed_minutes * 1000 * 60)) {
+        if (bg_missed_alerts && BgReading.getTimeSinceLastReading() > (bg_missed_minutes * 1000 * 60)
+                &&
+            prefs.getLong("alerts_disabled_until", 0) < new Date().getTime()) {
             Notifications.bgMissedAlert(mContext);
             checkBackAfterSnoozeTime();
-        } else {
-            checkBackAfterMissedTime();
+        } else  {
+            long alarmIn = (prefs.getLong("alerts_disabled_until", 0) - new Date().getTime());
+            if (alarmIn <= 0) {
+                alarmIn = Long.parseLong(prefs.getString("bg_missed_minutes", "30"));
```

Comment:
The behaviour should be the same as before that change (except in case all alerts are disabled off course) - correct me if I'm wrong.
Name of checkbackAfterMissedTime should actually be changed, better be checkBackAfter, because now it's only after missed time that a new check is done, it's also after alerts are re-enabled.

Is checkback after missedTime necessary ? probably not because the MissedReadingService is getting launched each time all alerts are checked (high alerts, low alerts), except again if all alerts are disabled.
You can see this in AlertType.java line 109, a call to checkifmissedreadingalert is made. That launches again the missedreadingService, and will reset the alarms in it. That's something I didn't change, but it's not very logic, although it works.